students/jared_mulholland/lesson_4/mail_room.py

```diff
@@ -0,0 +1,130 @@
+"""
+SENDING A THANKYOU
+If the user (you) selects ‘Send a Thank You’, prompt for a Full Name.
+    -If the user types ‘list’, show them a list of the donor names and re-prompt--
+    -If the user types a name not in the list, add that name to the data structure and use it.--
+    -If the user types a name in the list, use it.--
+    -Once a name has been selected, prompt for a donation amount.--
+    -Turn the amount into a number – it is OK at this point for the program to crash if someone types a bogus amount.--
+    -Once an amount has been given, add that amount to the donation history of the selected user. -- 
+    -Finally, use string formatting to compose an email thanking the donor for their generous donation. Print the email to the terminal and return to the original prompt.
+    -It is fine (for now) to forget new donors once the script quits running.--
+"""
+
+def send_thankyou():
+    name = input("Please Enter a Name: ")
+
+    while name == "list":
+        for donor in donors:
+            print(donor)
+        name = input("\nPlease Enter a Name: ")
+
+    if name not in donors:
+        donors[name] = []
+        new_donation = input("Enter a Donation Amount: ")
+        new_donation = float(new_donation)
+        #append new_donation to donor list
+        donors[name].append(new_donation)    
+        
+
+    elif name in donors:
+        new_donation = input("Enter a Donation Amount: ")
+        new_donation = float(new_donation)
```

Comment:
you can move above 2 lines out of these blocks to above first if check since they are duplicated.